lib/openapi/api/api_v1.go

```diff
@@ -259,13 +259,36 @@ func (e *Processor) ResourceAccessPut(c echo.Context, uid types.ResourceUID) err
 		return fmt.Errorf("Only the owner & admin can assign service mapping to the Application")
 	}
 
+	pwd := crypt.RandString(64)
+	pwdHash, err := crypt.NewHash(pwd, nil).Serialize()
+	if err != nil {
+		c.JSON(http.StatusBadRequest, H{"message": "Unable to prepare password hash"})
+		return fmt.Errorf("Unable to prepare password hash: %w", err)
+	}
+	key, err := crypt.GenerateSSHKey()
+	if err != nil {
+		c.JSON(http.StatusBadRequest, H{"message": "Unable to generate SSH key"})
+		return fmt.Errorf("Unable to generate SSH key: %w", err)
+	}
+	pubkey, err := crypt.GetSSHPubKeyFromPem(key)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, H{"message": "Unable to generate SSH public key"})
+		return fmt.Errorf("Unable to generate SSH public key: %w", err)
+	}
 	rAccess := types.ResourceAccess{
 		ResourceUID: res.UID,
 		Username:    user.Name,
-		Password:    crypt.RandString(64),
+		// We should not store clear password, so convert it to salted hash
+		Password: string(pwdHash),
+		// Key need to be stored as public key
+		Key: string(pubkey),
 	}
 	e.fish.ResourceAccessCreate(&rAccess)
 
+	// Overriding the password and key to return user the actual values one time and forget about
+	rAccess.Password = pwd
+	rAccess.Key = string(key)
```

Comment:
Yep, we are not storing those in DB - just returning to user so he can keep them. The comment says that we overriding the original values (pubkey/hash that are stored in DB) to return password/privkey to user.

Reply:
Ok, that was not clear to me.  I think updating the comment would make it easier to understand, maybe something like

```go
// Now database has had the hashed credentials stored, restore the original values
// to return to the user so they have access to the correct (un-hashed) credentials.
rAccess.Password = pwd
rAccess.Key = string(key)
```

the current comment (particularly the `one time and forget about` part) did not make it clear _why_ we were resetting the values.  I think the above would help future readers understand better, rather than seeing the comment a few lines above and think there is a mistake.

Reply:
Ok fixed it in daba843 and recalled why I needed port in ResourceAccess - to give user a way to find the port to connect to, so replaced it with Address to prepare for later in-cluster serving the request by another node (for example when it's needed to reach some particular local environment).